### PR TITLE
Move to CNF Cert Suite v5.1.1, including new way of launching the certsuite

### PR DESCRIPTION
**Pull request by ramperher**
Test-Hints: no-check

---------

- Move to tnf v5.1.1 as default.
- New certsuite tool is proposed to launch the tnf container. Moving to that approach for this version - https://github.com/test-network-function/cnf-certification-test/blob/main/cmd/certsuite/run/run.go
- Keeping the old way of launching the cert suite as backwards compatibility, to be removed at some point.
- Setting common steps for both ways of launching the certification suite (create config/result dir, put the files in the same place...)

---------

Tests:

- [x] tnf-test-cnf-green - unstable - passed - https://www.distributed-ci.io/jobs/3ef38548-2a14-447c-9065-c0bf5a85d24c/jobStates?sort=date
- [x] tnf-test-cnf - v5.1.1 - passed - https://www.distributed-ci.io/jobs/c4650a1c-736b-4eeb-a68d-8b212866f12c/jobStates?sort=date
- [x] tnf-test-cnf - v5.1.0 - passed - https://www.distributed-ci.io/jobs/1b1e860b-0726-4b54-aa09-6303133466a0/jobStates?sort=date

### CODE REVIEW

**Comment by ramperher** [comment]
Change is ready for review but I'll hold on the change till verifying what's happening with DeployableByOLM preflight test launched by tnf, which is now failing.

File: roles/cnf_cert/tasks/logging.yml
Comment (manurodriguez) [inline roles/cnf_cert/tasks/logging.yml:78]: LGTM, just to confirm. looks like in TNF v5.1.1 we only have `certsuite.log` log right? in the job with v5.1.0 version I see both, the `certsuite.log` and `dci-tnf-execution.log`

**Review by manurodriguez — DISMISSED** [review]

File: roles/cnf_cert/tasks/logging.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/logging.yml:78]: Yes, it was renamed at some point to `certsuite.log`, so, now that for v5.1.1 I'm proposing to directly use the container with `podman run`, then I've omitted the capture of logs in the file called `dci-tnf-execution.log`, that's why this file is not generated in 5.1.1, and with `certsuite.log`, it should be enough to check what we need to check in this task.
Also, there are some conditionals in the code to ensure backwards compatibility. When all consumers of this role move to, at least, 5.1.1, I'll start removing the support of the old code, as I did some months ago with 4.x.x.

File: roles/cnf_cert/tasks/logging.yml
Comment (reihl) [inline roles/cnf_cert/tasks/logging.yml:72]: Can we switch it to ternary?
"{{ (test_network_function_version is version("v5.1.0", "<=") and test_network_function_version != "HEAD") | ternary('dci-tnf-execution.log', 'certsuite.log')}}"

File: roles/cnf_cert/tasks/tests.yml
Comment (reihl) [inline roles/cnf_cert/tasks/tests.yml:75]: Why not use the podman module?

Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:75]: Could be used, but since running this podman container requires a bunch of arguments and parameter, I preferred to use the shell module as a first step, similarly to what we do with [preflight](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/roles/preflight/tasks/test_preflight_check_container_one_image.yml#L70-L96).
Let me finish the testing and I can try to explore that. Could be a good improvement here.

File: roles/cnf_cert/tasks/logging.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/logging.yml:72]: I started doing that in that way because I planned to use more than two values but finally I just used two alternatives, so yes, it's fine for me to move to ternary.

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:75]: @reihl , I tried here to use `podman` module: https://www.distributed-ci.io/jobs/2143a51b-307a-4fbe-8073-9f00b4b5c8ca/jobStates, but it's a bit tricky right now to do this transition. The execution doesn't like to work right now by using just exactly the same arguments I'm using with `shell`, and also, this would require a loop task to check the status of the container to confirm it's working fine.
I think it's better to keep this as it is right now, just to follow the trend that we already follow with preflight, as I said before, but since this role would require a refactor at some point, I'll take care of addressing that point in that moment. Is it fine for you?

Comment (reihl) [reply roles/cnf_cert/tasks/tests.yml:75]: Yes, we tried and face problems that we can avoid when we using shell. I just want to put this modules in our consideration. 
Having said that, can we consider using set_fact and convert the container command to a variable?

File: roles/cnf_cert/tasks/logging.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/logging.yml]: I've finally kept dci-tnf-execution.log file for tnf v5.1.1 too, so I've removed this new code and I'll keep using the old one.

File: roles/cnf_cert/tasks/logging.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/logging.yml]: I've finally kept dci-tnf-execution.log file for tnf v5.1.1 too, so I've removed this new code and I'll keep using the old one.

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:75]: Sure, I'll have the usage of podman module into consideration when addressing the next refactor of this role. I think it will be benefitial, but as I said, as this new way is also a fix for a current problem, and the integration of podman module would require to add more complexity to the role, I'd prefer to postpone this.

Regarding using set_fact, sure, I'll define the command in a variable (mainly with `vars`, as it's only used in this task) and give a try.

File: roles/cnf_cert/tasks/tests.yml
Comment (capolrik) [inline roles/cnf_cert/tasks/tests.yml:69]: *common

File: roles/cnf_cert/tasks/tests.yml
Comment (capolrik) [inline roles/cnf_cert/tasks/tests.yml:110]: If you are using the `no_log` option above, you may want to use it here as well.

File: roles/cnf_cert/defaults/main.yml
Comment (capolrik) [inline roles/cnf_cert/defaults/main.yml:6]: Is this variable still used?

Comment (ramperher) [reply roles/cnf_cert/defaults/main.yml:6]: Yes, if no `test_network_function_version` is provided in the pipeline, it's used this one by default: https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/roles/cnf_cert/tasks/pre-run.yml#L152-L169, which is what happens with `tnf-test-cnf` pipeline

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:110]: Not really required because the output cannot be seen in this task but I've done it just in case, thanks!

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:69]: fixed

**Review by tkrishtop — APPROVED** [review]
LGTM, thanks @ramperher

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml]: This is done, now the command is provided in a separate variable.

**Comment by ramperher** [comment]
Tests are finished and all feedback has been integrated, I'll wait for the `local/check` to be fixed by DCI team before merging, also waiting for some more comments and feedback.

File: roles/cnf_cert/tasks/tests.yml
Comment (tonyskapunk) [inline roles/cnf_cert/tasks/tests.yml:57]: Maybe like this since it's already inside `tnf_dir` ?
```suggestion
    ./run-tnf-container.sh -k config_files/kubeconfig \
```

File: roles/cnf_cert/tasks/tests.yml
Comment (tonyskapunk) [inline roles/cnf_cert/tasks/tests.yml:59]: Maybe?

```suggestion
    -t config_files \
    -o result_files \
```

**Review by tonyskapunk — APPROVED** [review]
LGTM

Left a couple of suggestions, don't forget to squash your changes before merging, thanks!

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:57]: I think I've been passing the arguments using the absolute path because I had issues with relative paths in the past. But I'll test this.

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml:59]: Same, I'll test this to confirm it works fine.

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml]: As commented in https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/331#issuecomment-2149295177, we cannot use relative paths for the arguments passed to `run-tnf-container`, so I will keep the code in the same status if you're fine with this.
I'll also keep `chdir: "{{ tnf_dir }}"` just to shortcut the call to `./run-tnf-container.sh` script, which is placed in that directory

File: roles/cnf_cert/tasks/tests.yml
Comment (ramperher) [reply roles/cnf_cert/tasks/tests.yml]: As commented in https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/331#issuecomment-2149295177, we cannot use relative paths for the arguments passed to `run-tnf-container`, so I will keep the code in the same status if you're fine with this.
I'll also keep `chdir: "{{ tnf_dir }}"` just to shortcut the call to `./run-tnf-container.sh` script, which is placed in that directory

**Comment by ramperher** [comment]
Change is ready for review again, launching again the DCI jobs to check everything is fine. Could you take a look at this? @tkrishtop @tonyskapunk thanks!

**Review by tonyskapunk — APPROVED** [review]
Thanks for taking a look and testing the suggestion, :+1:

**Comment by tonyskapunk** [comment]
recheck

**Pull request merged**
